### ISSUE TITLE
php::ensure Can I set the php version and specify latest patch release?

### ISSUE THREAD

**Issue by bobweston** [mayflower/puppet-php#187]
This might have an obvious answer, if so, please forgive my stupidity. :)
But is there a way to pin the php version while also allowing for the latest package release version.
I've set php::ensure to 5.5.9+dfsg-1ubuntu4.14
but as you might expect, that causes problems when trying to build later (After support for that patch release is dropped by Ubuntu).

5.5.9+dfsg-1ubuntu4.14 failed: Could not update

I fixed it to set the patch release to 5.5.9+dfsg-1ubuntu4.16.

But in spinning up my VM today, I'm getting the same error because Ubuntu has dropped that package release and is now using 5.5.9+dfsg-1ubuntu4.17.

Basically, I'd like to ensure that, going forward, I stay on php 5.5.9 and be on (at least the latest patch version). So it would seem setting php::ensure to latest would not be an option. (as I have some legacy php applications which will break if I move to anything later than 5.5.9)

Do I just need to set my hiera config to 

php::php_version: '5.5.9'

?
I really appreciate any help or suggestions you could provide.

Thank you.

**Comment by bobweston** [comment]
It appears that the solution to my issue is I needed to specify the version in hiera using php_version, not ensure.

php::php_version: '5.5.9'
